x-pack/metricbeat/module/googlecloud/compute/_meta/data.json

```diff
@@ -0,0 +1,77 @@
+{
+    "_index": "metricbeat-8.0.0-2019.11.27-000001",
+    "_id": "H4iRrm4B7vvv-udOwwFO",
+    "_version": 1,
+    "_score": null,
+    "_source": {
+        "@timestamp": "2019-11-27T20:30:00.000Z",
+        "ecs": {
+            "version": "1.2.0"
+        },
+        "host": {
+            "name": "mcastro"
+        },
+        "agent": {
+            "version": "8.0.0",
+            "type": "metricbeat",
+            "ephemeral_id": "cb56f25c-fc0e-4bbc-9c49-4551c1a97ab5",
+            "hostname": "mcastro",
+            "id": "7e36a073-1a32-4a94-b65b-4c7f971fb228"
+        },
+        "cloud": {
+            "availability_zone": "us-central1-a",
+            "account": {
+                "id": "elastic-metricbeat"
+            },
+            "provider": "googlecloud",
+            "instance": {
+                "id": "3296438752514919533",
+                "name": "instance-3"
+            }
+        },
+        "event": {
+            "module": "googlecloud",
+            "duration": 1252249339,
+            "dataset": "googlecloud.compute"
+        },
+        "metricset": {
+            "period": 300000,
+            "name": "compute"
+        },
+        "service": {
+            "type": "googlecloud"
+        },
+        "googlecloud": {
+            "compute": {
+                "labels": {
```

Comment:
I recommend to read the proposal to understand the problem of labelling in Google Cloud.

I don't see a _big_ inconvenience in placing labels outside of the metricset scope, but those labels are a mix of stackdriver and metricset labels so it would be incorrect to place "metricset" labels under the module scope. That's why I placed them all at the metricset level.

As I said, it's not a _big_ inconvenience, it was just a matter of correctness.

Reply:
ey @sayden do you have some examples of this? The point about having them together is normally that you can filter using the same label across different metricsets

Reply:
A simple example is `device_type`. It will appear in `googlecloud.labels.metrics.device_type`. Well... that's a bit out of context and its correct place would be `googlecloud.compute.metrics.device_type`. Actually, the `label_key` and `metadata_key` key-values that you can see in the JSON, are also Compute specific labels that were inserted in the Compute console by myself. So they might be referred to a compute resource like `production_cluster` and it will appear in `googlecloud.labels.metadata.production_cluster`... weird, isn't it? :smile: You might have Dataproc production clusters and Compute production clusters.

We don't have (yet) any global field that is not being captured by the ECS "soft-middleware" tu put as a example, but I also see the problem of having a `googlecloud.compute.region` field that I want to group with `googlecloud.pubsub.region`. As I said, that's not a problem right now but I see the point and I don't have any strong opinion about it.

Reply:
I'll wait for your decision :)

Reply:
I think that at least user defined labels should be stored in a common place, if I tag some of my resources with some label I would expect to be able to filter their metrics, logs and so on using these labels.

If there are also service-specific labels, then I am ok with leaving them in service-specific fields.

Reply:
Placed into the module namespace. I still need to generate an example to upload.